File: app/src/main/java/com/example/shoppinglist/viewModels/ShoppingListViewModel.java
package com.example.shoppinglist.viewModels;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import com.example.shoppinglist.models.ShoppingList;
import com.example.shoppinglist.repositories.ShoppingListRepository;

import java.util.List;

public class ShoppingListViewModel extends AndroidViewModel {
    private ShoppingListRepository repository;
    private LiveData<List<ShoppingList>> allCurrentShoppingList;
    private LiveData<List<ShoppingList>> allArchivedShoppingList;

    public ShoppingListViewModel(@NonNull Application application) {
        super(application);
        repository = new ShoppingListRepository(application);
        allCurrentShoppingList = repository.getAllCurrentShoppingList();
        allArchivedShoppingList = repository.getAllArchivedShoppingList();
    }

    public void insertShoppingList(ShoppingList shoppingList) {
        repository.insertShoppingList(shoppingList);
    }

    public void updateShoppingLis(ShoppingList shoppingList) {
        repository.updateShoppingList(shoppingList);
    }

    public void deleteShoppingList(ShoppingList shoppingList) {
        repository.deleteShoppingList(shoppingList);
    }

    public LiveData<List<ShoppingList>> getAllCurrentShoppingList() {
        return allCurrentShoppingList;
    }

    public LiveData<List<ShoppingList>> getAllArchivedShoppingList() {
        return allArchivedShoppingList;
    }
}

File: app/src/main/java/com/example/shoppinglist/viewModels/ShoppingListViewModel.kt
package com.example.shoppinglist.viewModels

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import com.example.shoppinglist.models.ShoppingList
import com.example.shoppinglist.repositories.ShoppingListRepository

class ShoppingListViewModel(application: Application) : AndroidViewModel(application) {

    private val repository: ShoppingListRepository = ShoppingListRepository(application)
    val allCurrentShoppingList: LiveData<List<ShoppingList>>
    val allArchivedShoppingList: LiveData<List<ShoppingList>>
    var insertionId: MutableLiveData<Long> = repository.getInsertionId()

    fun insertShoppingList(shoppingList: ShoppingList?) {
        repository.insertShoppingList(shoppingList!!)
    }

    fun updateShoppingLis(shoppingList: ShoppingList?) {
        repository.updateShoppingList(shoppingList!!)
    }

    fun deleteShoppingList(shoppingList: ShoppingList?) {
        repository.deleteShoppingList(shoppingList!!)
    }

    fun deleteEmptyShoppingList() {
        repository.deleteEmptyShoppingList()
    }

    init {
        allCurrentShoppingList = repository.getAllCurrentShoppingList()
        allArchivedShoppingList = repository.getAllArchivedShoppingList()
    }
}
